### DISCUSSION TITLE
Name Change

### DISCUSSION THREAD

**Discussion by ghost** [fastapi/fastapi#8125] [Questions] [answered]
Hi, this might seem like nitpicking, but I feel very strongly that a big part of what leads a project to be successful, specially amidst the competitiveness of web frameworks, is the way it spreads throughout the community.

I would consider the name of this project to possibly be detrimental to it's success, FastAPI doesn't roll like a brand at all, just a worry. :)

**Comment by jet10000** [comment]
In addition to the framework itself, it also needs to be combined with other projects, such as traditional web, current spa projects, back-end various types of databases, newcomers' tutorials, best practice documentation, and video tutorials.  Pay attention to how I use it, what value it can bring to my project. More than just technology, there are real solutions.

Currently I think it's very attractive to django developers and can break through some of the limitations of django.  For django developers to migrate and use them in new projects, you must also have the convenience of django.

It’s hard for people in the javascript community to migrate to FastAPI.Because they have a lot of similar things.

Any project needs more people to use, and let django users use fastapi is a huge opportunity.

**Comment by EthanC** [comment]
I second this, I've been meaning to open an issue regarding it for some time now. FastAPI is an extraordinary framework, but it could use a name change to something more unique and identifiable.

If/when FastAPI becomes mainstream, it's going to be quite difficult for a new developer to Google their issue. "Python Fast API authentication" isn't going to return results as perfect as something like "Python Django authentication" because "FastAPI" isn't a unique term.

On top of this, if FastAPI is to be used in projects which are hiring, "Minimum 2 years experience with FastAPI" just doesn't look as good as "Minimum 2 years experience with Flask".

CC @tiangolo ❤

**Comment by tiangolo** [comment]
I'm glad you are liking it!

> FastAPI doesn't roll like a brand at all

Hmm, I guess it depends on the point of view. I like that it gives away what it is about. And it seems it hasn't been a problem for it to be shared by the community. I think about 3K stars in like 6 months is quite good :smile: Also, there are articles about it in English, Mandarin, Japanese, Russian, Vietnamese, etc. So I think it seems to be working well. :tada: 

> [...] a new developer to Google their issue [...]

I'm pretty sure if you Google `python fastapi authentication` you will get some of the docs in the top results, plus some articles about FastAPI, docs in OAuth2.net, etc.

The same way, search in Twitter `fastapi`, there's quite some stuff there...

> [...] "Minimum 2 years experience with FastAPI" just doesn't look as good as "Minimum 2 years experience with Flask" [...]

Hmm, I guess that's subjective. I've seen some job offers wanting FastAPI experience already, I don't think it's a problem. :wink: :money_with_wings: 

Also ...I kinda like the name FastAPI :grin:

**Comment by cesarandreslopez** [comment]
I suggest it just gets called TiangoAPI.

Just an idea :)

**Comment by varunchopra** [comment]
@tiangolo I feel that you're really underselling yourself. The work done here is easily worth a lot more than 3k stars in 6 months.

I agree with OP. A name change will improve the SEO by a lot. I've had issues finding relevant content using search engines too. The only way to find good content is by checking out the chat where users share their blog posts sometimes.

**Comment by cesarandreslopez** [comment]
@chopraaa just curious. Do you have a suggestion on a name?

There is already some amount of positioning in the name "FastAPI"

**Comment by varunchopra** [comment]
@cesarandreslopez  Don't have the slightest idea 😄 . In the end it's up to Sebastian I guess.

A quick search on DuckDuckGo reveals this: https://duckduckgo.com/?q=fast+api&t=hj&ia=web Google is a bit more generous, giving 2 links to this project but it's still not enough.

The general idea is to come up with a term that isn't this common, so that users looking for Django and Flask can come across this without explicitly knowing the term 'FastAPI'

**Comment by dmontagu** [comment]
For what it's worth, I think https://duckduckgo.com/?q=fastapi&t=hj&ia=web is a better reference -- nearly all of the top results are related to this project. Not to say that no one would ever search for "fast api" instead of "fastapi", but I think it would be a bigger deal if searching the name proper had the same results. My experience has been that SEO hasn't been a big problem while searching for answers as long as I search "fastapi" instead of "fast api".

**Comment by stefanondisponibile** [comment]
When I first came across it, fastAPI caught my attention for its name. It sounds and looks modern.

**Comment by euri10** [comment]
I would suggest spend less time discussing a name change (which is like
colors, everyone likes a different one) and more helping others, reviewing
bugs, answer questions or developing features, which in the end is what's
everyone truly wants.
Whatever good name you may have, if you don't have the latter then what's
the point?


Le lun. 24 juin 2019 à 11:59 PM, Stefano <notifications@github.com> a
écrit :

**Comment by tiangolo** [comment, marked as answer]
Thanks for the feedback guys!

There's also the quite important issue that if I changed the name now, many applications would break.

Many articles and blog posts would be obsolete, and everyone that already knows the name "FastAPI"(good or bad) would be lost and would have to start and search again, so the problem of "spreading throughout the community" would actually be increased, even if there was a much better name.

Also, names are quite a subjective topic, it's easy to end up ["bikeshedding"](https://en.wiktionary.org/wiki/bikeshedding).

So I think I won't change the name easily, to avoid those problems.

An extreme case would be that there was a standard/spec that already used the name in a similar context and this name only lead to confusion (e.g. if the framework was named `openapi`, like what happened for the package `pep8`).

Other extreme case was that the name was very explicitly offensive for a group of people or similar.

---

I'll close this issue now. But you can add more comments/arguments here in this thread or your name suggestions if you want. But apart from those or similar cases, I think a name change is not very feasible.

Again, thanks for the feedback everyone!
